### PR TITLE
Bug fix: Replace prop_cycler with Cycler()

### LINKED ISSUE #3346: [Bug]: failing example notebook and script for COMSOL discharge curves
### PyBaMM Version

develop

### Python Version

3.9

### Describe the bug

The example notebook `models/compare-comsol-discharge-curve.ipynb` and the script `examples/scripts/compare_comsol/discharge_curve.py` are failing in some workflow runs. 

Also, the example scripts were not being tested in the scheduled tests. I have opened a PR at #3345 to fix that

The links to the error logs are here: 

1. https://github.com/pybamm-team/PyBaMM/actions/runs/6204934533/job/16847574605#step:14:173
2. https://github.com/pybamm-team/PyBaMM/actions/runs/6208214736/job/16854166967?pr=3344#step:9:188
3. This is also noted in https://github.com/pybamm-team/PyBaMM/pull/3335#issuecomment-1721391033 and also while @arjxn-py and I were working on #3344.

### Steps to Reproduce

Run the example notebook and script with the latest changes from the `develop` branch

### Relevant log output

_No response_

### PR DESCRIPTION
**Pull request by kratman**
# Description

Matplotlib updated to 3.8 on Friday. Since then the line
```python
color = next(ax._get_lines.prop_cycler)["color"]
```
causes the following error:
```
AttributeError: '_process_plot_var_args' object has no attribute 'prop_cycler'
```

I have replaced the prop_cycler with a cycler object and configured it with a generic color order. This was the most minimal change I could think of to resolve the issue. I am open to suggestions for a specific color set or I can just remove the explicit cycler and let matplotlib handle the colors itself. The latter also resolves the issue.

Fixes #3346 

## Type of change

- [ ] New feature (non-breaking change which adds functionality)
- [ ] Optimization (back-end change that speeds up the code)
- [x] Bug fix (non-breaking change which fixes an issue)

# Key checklist:

- [x] No style issues: `$ pre-commit run` (or `$ nox -s pre-commit`) (see [CONTRIBUTING.md](https://github.com/pybamm-team/PyBaMM/blob/develop/CONTRIBUTING.md#installing-and-using-pre-commit) for how to set this up to run automatically when committing locally, in just two lines of code)
- [x] All tests pass: `$ python run-tests.py --all` (or `$ nox -s tests`)
- [x] The documentation builds: `$ python run-tests.py --doctest` (or `$ nox -s doctests`)

You can run integration tests, unit tests, and doctests together at once, using `$ python run-tests.py --quick` (or `$ nox -s quick`).

## Further checks:

- [x] Code is commented, particularly in hard-to-understand areas
- [x] Tests added that prove fix is effective or that feature works

### CODE REVIEW

**Comment by kratman** [comment]
@agriyakhetarpal Here is part 1, I expect tests to still fail since there are multiple issues.

File: examples/scripts/compare_comsol/discharge_curve.py
Comment (kratman) [inline examples/scripts/compare_comsol/discharge_curve.py:1]: The changes in the notebook are the same in this file. The notebook and script share a lot of the same code.

**Review by brosaplanella — APPROVED** [review]
Looks good, thanks!

**Comment by brosaplanella** [comment]
@all-contributors please add @kratman for bug fixes

**Pull request merged**